backend/runtimestate/runtimestate.go

```diff
@@ -140,8 +140,35 @@ func ApplyActions(s *protos.OrchestrationRuntimeState, customStatus *wrapperspb.
 					Router: action.Router,
 				})
 				if s.StartEvent.GetParentInstance() != nil {
+					// Create a router for the completion event that routes back to the parent
+					var completionRouter *protos.TaskRouter
+					if action.Router != nil {
+						var parentAppID string
+
+						allEvents := append(s.OldEvents, s.NewEvents...)
+						for _, event := range allEvents {
+							if es := event.GetExecutionStarted(); es != nil && event.GetRouter() != nil {
+								parentAppID = event.GetRouter().GetSource()
+								break
+							}
```

Comment:
Is it ever the case that there are multiple `GetExecutionStarted`? Should we not have a `source`/`router` type field on the `ParentInstance` to determine source app ID rather than searching in the event history?

Reply:
Makes sense, opened [this PR](https://github.com/dapr/durabletask-protobuf/pull/16) adding this optional field.